packages/unit/src/main.ts

```diff
@@ -23,45 +24,58 @@ const SuitesDIAdapters = {
-    .resolveCorrespondingAdapter()
-    .then((adapter) => adapter)
-    .catch(() => {
-      throw new AdapterNotFoundError(`Suites requires an adapter to integrate with different dependency injection frameworks.
+      const diAdapter = diPackageResolver
+        .resolveCorrespondingAdapter()
+        .then((adapter) => adapter)
+        .catch(() => {
+          throw new AdapterNotFoundError(`Suites requires an adapter to integrate with different dependency injection frameworks.
 It seems that you haven't installed an appropriate package. To resolve this issue, please install
 one of the available packages that matches your dependency injection framework.
 Refer to the docs for further information: https://suites.dev/docs`);
-    });
+        });
 
-  const doublesPackageResolver = new PackageResolver<MockFunction<unknown>>(doublesAdapters);
+      const doublesPackageResolver = new PackageResolver<MockFunction<unknown>>(doublesAdapters);
 
-  const doublesAdapter = doublesPackageResolver
-    .resolveCorrespondingAdapter()
-    .then((adapter) => adapter)
-    .catch(() => {
-      throw new AdapterNotFoundError(`Suites requires an adapter to integrate with different mocking libraries.
+      const doublesAdapter = doublesPackageResolver
+        .resolveCorrespondingAdapter()
+        .then((adapter) => adapter)
+        .catch(() => {
+          throw new AdapterNotFoundError(`Suites requires an adapter to integrate with different mocking libraries.
 It seems that you haven't installed an appropriate adapter package. To resolve this issue, please install
 one of the available packages that matches your mocking library.
 Refer to the docs for further information: https://suites.dev/docs`);
-    });
+        });
 
-  const unitMocker = new UnitMocker(doublesAdapter);
-  return new TestBedBuilder<TClass>(doublesAdapter, diAdapter, unitMocker, targetClass, console);
+      const unitMocker = new UnitMocker(doublesAdapter, diAdapter);
+
+      return new type(doublesAdapter, diAdapter, unitMocker, targetClass, console) as TBuilder;
+    },
+  };
 }
 
 export class TestBed {
-  public static solitary<TClass = any>(targetClass: Type<TClass>): TestBedBuilder<TClass> {
-    return createTestbedBuilder(SuitesDIAdapters, SuitesDoublesAdapters, targetClass);
+  public static solitary<TClass = any>(targetClass: Type<TClass>): SolitaryTestBedBuilder<TClass> {
+    return testBedBuilderFactory(SuitesDIAdapters, SuitesDoublesAdapters, targetClass).create(
+      SolitaryTestBedBuilder
+    );
+  }
+
+  public static sociable<TClass = any>(targetClass: Type<TClass>): SociableTestBedBuilder<TClass> {
+    return testBedBuilderFactory(SuitesDIAdapters, SuitesDoublesAdapters, targetClass).create(
+      SolitaryTestBedBuilder
+    );
   }
 
-  public static create<TClass = any>(targetClass: Type<TClass>): TestBedBuilder<TClass> {
+  public static create<TClass = any>(targetClass: Type<TClass>): SolitaryTestBedBuilder<TClass> {
```

Comment:
Backwards-compatible API. Awesome!

Reply:
lol. Yes it'a gonna be a whole new framework under the scope `@suites`.